### PR TITLE
doc/start: rewrite hardware docs - CPU section

### PR DESCRIPTION
**Pull request by zdover23**
This PR updates the CPU section of the "Hardware
Recommendations" page for the first time since
2014.

Signed-off-by: Zac Dover <zac.dover@gmail.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: doc/start/hardware-recommendations.rst
Comment (tchaikov) [inline doc/start/hardware-recommendations.rst:25]: i don't follow. "CPU intensive" means large memory footprint?

Comment (zdover23) [reply doc/start/hardware-recommendations.rst:25]: I'll take it out.

**Review by dvanders — APPROVED** [review]

**Review by markhpc — CHANGES_REQUESTED** [review]

File: doc/start/hardware-recommendations.rst
Comment (anthonyeleven) [inline doc/start/hardware-recommendations.rst:32]: FWIW, in 2014 I commonly heard 1 core/OSD.  HT wasn't usually explicitly mentioned, but was usually encouraged, so that was effectively 1 *thread* per OSD.

Comment (zdover23) [reply doc/start/hardware-recommendations.rst:32]: @markhpc, I'm going to defer to you on this one. What do you say?

Comment (markhpc) [reply doc/start/hardware-recommendations.rst:32]: Sorry for the delay, agreed 100%.  2 cores per OSD is easily enough for HDDs, but for flash storage it's just one point (that's fairly low) on the performance curve.  Certainly enough to make the OSD function (though you can probably function with 1 core as well), but nowhere near where we top out if pursuing performance with fast devices.

Comment (anthonyeleven) [reply doc/start/hardware-recommendations.rst]: Agreed.

**Review by dvanders — APPROVED** [review]

**Comment by mheler** [comment]
Is there a recommendation for SSD and HDD's as well?

File: doc/start/hardware-recommendations.rst
Comment (anthonyeleven) [inline doc/start/hardware-recommendations.rst:29]: Suggest:
s/metadata servers/metadata servers (MDS)/

s/therefore..Ghz)/have a high clock frequency (GHz) CPU model. For MDS purposes, higher _frequency_ cores are more beneficial than a large _number_ of cores.  Four or eight cores should be plenty for a server that only hosts the MDS service.  If you must colocate other services, eg. Mon, manager, or OSDs, a larger number of cores is advised.

s/need hardware with/need/

s/map/map.  More cores are beneficial for OSD services; we suggest at least two cores per OSD.  NVMe OSDs may benefit from additional cores, and a higher clock frequency may decrease op latency on these very fast devices as well./

**Comment by anthonyeleven** [comment]
> Is there a recommendation for SSD and HDD's as well?

That's a bit of a rabbit hole, but OSD drives are mentioned at the tail end of the same file. And I *think* elsewhere in the doc tree?  I remember us recent adding a reference to the SNIA TCO calculator somewhere.

 There are indeed some updates I'd like to make there once this PR is merged.
* Note the bottleneck of large (8+ TB) SATA HDDs
* Mention QLC for suitable workloads as an HDD alternative
* Note the cost of HBA/BBU when considering HDDs

**Comment by zdover23** [comment]
This PR is superseded by https://github.com/ceph/ceph/pull/47109

**Pull request closed without merging**
